pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go

```diff
@@ -144,6 +155,12 @@ func (e *AzureLogAnalyticsDatasource) buildQueries(ctx context.Context, queries
 				resourceOrWorkspace = *azureLogAnalyticsTarget.Workspace
 			}
 
+			meetsBasicLogsCriteria, meetsBasicLogsCriteriaErr := MeetsBasicLogsCriteria(basicLogsQueryflag, resources, fromAlert)
```

Comment:
I suggest restructuring this so it only runs if `basicLogsQueryFlag` is `true` to avoid running this for every query

Reply:
I changed this! Let me know what you think